lib/logger.js

```diff
@@ -638,6 +645,12 @@ class Logger extends EventEmitter {
     , httpAgent: undefined
     }
 
+    // Setting this to `false` can avoid the browser console error of:
+    // Refused to set unsafe header "user-agent"
+    if (this.sendUserAgent) {
+      config.headers['user-agent'] = this[kUserAgentHeader]
+    }
```

Comment:
Why check and set this for each request, instead of doing the check/set on the defaults?

Reply:
Good question.   Axios used to not even allow custom `user-agent` headers, but when they added the "fix", there were reports that the same fix did not apply to the defaults.  Since we have to build this object for each request anyway, it wouldn't really be any better to try and use the defaults for it.